ofrak_core/test_ofrak/unit/test_ofrak_server.py

```diff
@@ -379,3 +394,267 @@ async def test_add_tag(ofrak_client: TestClient, hello_world_elf):
+    children_body = await children_resp.json()
+
+    child_one = children_body[root_id][0]
+    child_two = children_body[root_id][1]
+    child_one_res = await ofrak_client.post(f"/{child_one['id']}/unpack")
+    child_two_res = await ofrak_client.post(f"/{child_two['id']}/unpack")
+    child_one_body = await child_one_res.json()
+    child_two_body = await child_two_res.json()
+    child_one_resources = child_one_body["created"]
+    child_two_resources = child_two_body["created"]
+    # Must sort results of unpacking so that comparison works
+    child_one_resources.sort(key=get_child_sort_key)
+    child_two_resources.sort(key=get_child_sort_key)
+
+    # Verify results of unpacking are the same for both children and are what we expect
+    attrs_to_skip = {"id", "data_id", "parent_id", "attributes"}
+    assert len(child_one_resources) == len(child_two_resources)
+    assert all(
+        list(
+            map(
+                dicts_are_similar,
+                child_one_resources,
+                child_two_resources,
+                itertools.repeat(attrs_to_skip),
+            )
+        )
+    )
+    assert all(
+        list(
+            map(
+                dicts_are_similar,
+                child_two_resources,
+                child_one_resources,
+                itertools.repeat(attrs_to_skip),
+            )
+        )
+    )
+    assert "ofrak.core.elf.model.ElfBasicHeader" in child_one_resources[0]["tags"]
+
+    # Verify script is as expected
+    resp = await ofrak_client.get(
+        f"/{root_id}/get_script",
+    )
+    resp_body = await resp.json()
+    assert resp_body == [
+        "from ofrak import *",
+        "from ofrak.core import *",
+        "",
+        "",
+        "async def main(ofrak_context: OFRAKContext):",
+        "",
+        "    root_resource = await ofrak_context.create_root_resource_from_file(",
+        '        "hello_world_elf"',
+        "    )",
+        "",
+        "    await root_resource.create_child(",
+        "        tags=(GenericBinary,), data_range=Range(0x0, 0x1FF5)",
+        "    )",
+        "",
+        "    await root_resource.create_child(",
+        "        tags=(GenericBinary,), data_range=Range(0x0, 0x1FF5)",
+        "    )",
+        "",
+        "    # Resource with parent root_resource is missing, could not find selectable attributes.",
+        "    raise RuntimeError(",
+        '        "Resource with ID 0x00000002 cannot be uniquely identified by attribute Data.Offset (resource has value 0)."',
+        "    )",
+        "",
+        "    await root_resource_MISSING_RESOURCE_0.unpack()",
+        "",
+        "    # Resource with parent root_resource is missing, could not find selectable attributes.",
+        "    raise RuntimeError(",
+        '        "Resource with ID 0x00000003 cannot be uniquely identified by attribute Data.Offset (resource has value 0)."',
+        "    )",
```

Comment:
@rbs-jacob that makes sense. One issue I thought of in particular is what happens if the resource ID changes. At the moment they appear to be stable across runs, but will that always be the case?

Reply:
That's a good point – they may not be stable across runs.

I'd add a comment that links to this thread, and suggests adding a "normalization" step wherein the test case reference and `resp_body` are canonicalized into a consistent format. 

Normalization/canonicalization would involve both the reference string and `resp_body` being joined into strings (instead of lists of strings) and then some regular expression replacements could be performed. For example, in this case, the replacement would be something like:

``` python
def normalize(list_of_strs):
    in_str = "\n".join(list_of_strs)
    return re.sub(r"RuntimeError(\n.*\n)", "RuntimeError(err)", in_str, re.M)
```